src/Database/LSMTree/Monoidal.hs

```diff
@@ -672,6 +673,24 @@ union :: forall m k v.
   -> m (Table m k v)
 union = error "union: not yet implemented" $ union @m @k @v
 
+{-# SPECIALISE unions ::
+     ResolveValue v
+  => V.Vector (Table IO k v)
+  -> IO (Table IO k v) #-}
+-- | Like 'union', but for @n@ tables.
+--
+-- A good mental model of this operation is @'Data.Map.Lazy.unionsWith' (<>)@ on
+-- @'Data.Map.Lazy.Map' k v@.
+--
+-- Exceptions:
+--
+-- * Unioning @<2@ tables is an exception.
```

Comment:
I figured because `unions` on 1 table are somewhat of a degenerate case, because then they should really be using `duplicate`. But I suppose we can be lenient here and allow `>=1` tables, because there is nothing that can go wrong in the implementation. I'll change that, and put a comment on `unions` to preferably use `duplicate` if `n=1`